lib/src/components/NxNumberInput/NxNumberInput.tsx

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import React, { forwardRef, FormEvent, KeyboardEvent, useRef, MutableRefObject, useMemo } from 'react';
+import classnames from 'classnames';
+import { omit } from 'ramda';
+import { faExclamationCircle, faCheck } from '@fortawesome/free-solid-svg-icons';
+
+import '../NxTextInput/NxTextInput.scss';
+import './NxNumberInput.scss';
+
+import NxFontAwesomeIcon from '../NxFontAwesomeIcon/NxFontAwesomeIcon';
+import { Props, propTypes, NumberInputElement } from './types';
+import { hasValidationErrors, getFirstValidationError } from '../../util/validationUtil';
+import { getUniqueId } from '../../util/idUtil';
+export { Props, PublicProps, StateProps, propTypes } from './types';
+
+/**
+ * Standard number input with validation styling
+ * @param value The value rendered in the input
+ * @param isPristine Should be set to true when the user has not yet adjusted the value of the input
+ * @param validationErrors Zero or more validation error messages.  If empty or not defined, the field is
+ *   considered to be valid
+ * @param onChange A callback for when the user changes the value of the text box (e.g. by typing a letter)
+ * @param onKeyPress A callback for when the user presses a key that doesn't necessarily change the input value
+ *    (e.g. by hitting enter)
+ */
+const NxNumberInput = forwardRef<HTMLDivElement, Props>(
+    function NxNumberInput(props, forwardedRef) {
+      const {
+        isPristine,
+        validatable,
+        validationErrors,
+        onChange,
+        className,
+        onKeyPress,
+        disabled,
+        prefixContent,
+        ...attrs
+      } = props;
+
+      /**
+       * `trimmedValue` is a hidden property in `props`
+       * We need to remove it so react doesn't complain when we pass the object
+       * with the props to the `createElement` method below.
+       * `title` is removed to prevent errors when `NxNumberInput`, which itself uses` NxTooltip`, is wrapped
+       * in another `NxTooltip`
```

Comment:
This doesn't seem true

Reply:
> This doesn't seem true

Gone as part of the refactoring here: https://github.com/sonatype/sonatype-react-shared-components/pull/486/commits/517924162c67ef8f5716a3c816aefe18f28357de